Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -189,82 +195,117 @@ public function render()
 
         foreach ($this->_aActions as $sActionType => $aSingleActionConfig) {
-            if ($oConfig->getRequestParameter($sActionType)) {
+            if ($oRequest->getRequestParameter($sActionType)) {
                 $aActionConfig = $aSingleActionConfig;
             }
         }
 
+        $aOrderArticles = array_combine(
+            $oRequest->getRequestParameter('article-number') ?? [],
+            $oRequest->getRequestParameter('quantity') ?? []
+        );
+
         return [
-            self::KEY_AMOUNT => Helper::getFloatFromString($oConfig->getRequestParameter(self::KEY_AMOUNT) ?? ''),
+            self::KEY_AMOUNT => $aOrderArticles ? null
+                : Helper::getFloatFromString($oRequest->getRequestParameter(self::KEY_AMOUNT) ?? ''),
             self::KEY_ACTION => $aActionConfig,
+            self::KEY_ORDER_ITEMS => $aOrderArticles,
         ];
     }
 
     /**
      * Validates a request.
      *
      * @param array $aRequestParameters
+     *
      * @throws StandardException
      *
      * @since 1.1.0
      */
     private function _validateRequest($aRequestParameters)
     {
+        $aOrderItems = $aRequestParameters[self::KEY_ORDER_ITEMS];
+
+        if ($aOrderItems) {
+            $this->_validateOrderItems($aOrderItems);
+        }
+
         $fAmount = $aRequestParameters[self::KEY_AMOUNT];
 
+        if (!is_null($fAmount)) {
+            $this->_validateAmount($fAmount);
+        }
+    }
+
+    /**
+     * Check for correct amount
+     *
+     * @param float $fAmount
+     *
+     * @throws StandardException
+     *
+     * @since 1.2.0
+     */
+    private function _validateAmount($fAmount)
+    {
         if (!$this->_isAmountNumeric($fAmount)) {
             throw new StandardException(Helper::translate('wd_text_generic_error'));
         }
 
         $sTransactionId = $this->_oTransaction->wdoxidee_ordertransactions__transactionid->value;
         $fMaxAmount = $this->_getTransactionHandler()->getTransactionMaxAmount($sTransactionId);
 
-        if (!$this->_isPositiveBelowMax($fAmount, $fMaxAmount)) {
+        if (!Helper::isPositiveBelowMax($fAmount, $fMaxAmount)) {
             throw new StandardException(Helper::translate('wd_total_amount_not_in_range_text'));
         }
     }
 
     /**
-     * Checks that the amount argument is a numeric value
      *
-     * @param float $fAmount
+     * Checks order items quantity
      *
-     * @return boolean true if $fAmount is a numeric value
+     * @param array $aOrderItems
      *
-     * @since 1.1.0
+     * @throws StandardException if all order items are zero
+     *
+     * @return void
+     *
+     * @since 1.2.0
      */
-    private function _isAmountNumeric($fAmount)
+    private function _validateOrderItems($aOrderItems)
     {
-        return $fAmount && (is_float($fAmount) || is_int($fAmount));
+        foreach ($aOrderItems as $sArticleNumber => $iQuantity) {
+            if ($iQuantity > 0) {
+                return;
```

Comment:
That will return after the first item is above 0 quantity, what if the second item is bellow? will it get checked?

Reply:
then it is fine to do the transaction. It only checks that there is at least one item in the order to have a transaction with.